src/main/java/com/zendesk/maxwell/producer/AbstractAsyncProducer.java

```diff
@@ -36,14 +34,12 @@ public void markCompleted() {
 
 				if (message != null) {
 					context.setPosition(message.position);
-					timeSinceSendMS = message.timeSinceSendMS();
+					Metrics metrics = context.getMetrics();
+					Timer timer = metrics.getRegistry().timer(metrics.metricName("message", "publish", "time"));
```

Comment:
We'd be able to reuse the same `timer` here.

Reply:
What do you mean? Can you elaborate pls? The code is actually moved from `MaxwellKafkaProducer`.